x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -133,137 +154,132 @@ export class ManifestManager {
     // Get diffs
     const diffs = newManifest.diff(oldManifest);
 
-    // Create new artifacts
-    for (const diff of diffs) {
-      if (diff.type === 'add') {
-        const artifact = newManifest.getArtifact(diff.id);
-        try {
-          await this.artifactClient.createArtifact(artifact);
-          // Cache the body of the artifact
-          this.cache.set(diff.id, artifact.body);
-        } catch (err) {
-          if (err.status === 409) {
-            // This artifact already existed...
-            this.logger.debug(`Tried to create artifact ${diff.id}, but it already exists.`);
-          } else {
-            throw err;
-          }
-        }
-      }
-    }
-
     return {
       manifest: newManifest,
       diffs,
     };
   }
 
   /**
-   * Dispatches the manifest by writing it to the endpoint packageConfig.
+   * Syncs artifacts based on provided snapshot.
+   *
+   * Creates artifacts that do not yet exist and cleans up old artifacts that have been
+   * superceded by this snapshot.
    *
-   * @return {WrappedManifest | null} WrappedManifest if all dispatched, else null
+   * Can be filtered to apply one or both operations.
+   *
+   * @param snapshot
+   * @param diffType
    */
-  public async dispatch(wrappedManifest: WrappedManifest | null): Promise<WrappedManifest | null> {
-    if (wrappedManifest === null) {
-      this.logger.debug('wrappedManifest was null, aborting dispatch');
-      return null;
-    }
+  public async syncArtifacts(snapshot: ManifestSnapshot, diffType?: 'add' | 'delete') {
```

Comment:
nit: would maybe make diffType a type and have `add` and `delete` as their own types as well.